tests/50federation/30room-join.pl

```diff
@@ -953,7 +953,110 @@ sub assert_is_valid_pdu {
 
             # XXX currently synapse fails with a 500 here. I'm not really convinced that's
             # a thing we want to enforce, but we don't really have a specced way to say
-            # "a remote server did something wierd".
+            # "a remote server did something weird".
+            Future->done(1);
+         }),
+      )
+   };
+
+test "Outbound federation event with an invalid signature should not break",
```

Comment:
I'd suggest being more specific about where we are inserting the broken event:

```suggestion
test "Event with an invalid signature in the send_join response should not cause room join to fail",
```